src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -396,23 +402,26 @@ case class WorkflowActor(workflow: WorkflowDescriptor, backend: Backend)
                                   hash: Option[ExecutionHash],
                                   resultsClonedFrom: Option[BackendCall],
                                   data: WorkflowData): State = {
-    logger.debug(s"handleCallCompleted for ${callKey.tag}")
     val currentSender = sender()
     val completionWork = for {
       // TODO These should be wrapped in a transaction so this happens atomically.
       _ <- globalDataAccess.setOutputs(workflow.id, callKey, callOutputs, callKey.scope.rootWorkflow.outputs)
       _ <- globalDataAccess.setExecutionEvents(workflow.id, callKey.scope.fullyQualifiedName, callKey.index, callKey.attempt, executionEvents)
-      _ = persistStatusThenAck(callKey, ExecutionStatus.Done, currentSender, message, Option(callOutputs), Option(returnCode), hash, resultsClonedFrom)
-    } yield()
+    } yield ()
 
-    completionWork onFailure {
-      case e =>
+    completionWork onComplete {
+      case Failure(e)=>
         logger.error(s"Completion work failed for call ${callKey.tag}! " + e.getMessage, e)
-        self ! PersistenceFailed(callKey, ExecutionStatus.Done)
+        self ! CallFailedNonRetryable(callKey, executionEvents, Option(returnCode), e.getMessage)
+      case Success(_) =>
+        self ! PersistStatus(callKey, ExecutionStatus.Done, Option(callOutputs), Option(returnCode), hash, resultsClonedFrom)
     }
 
-    val updatedData = data.addPersisting(callKey, ExecutionStatus.Done)
-    stay() using updatedData
+    /* Ack now to the Call Actor that its message is being processed,
+     * we don't need to actually wait for the status to be persisted as this is internal to the WA processing algorithm.
+     */
```

Comment:
@mcovarr thinking more about this, I think it's actually ok to send the Ack to the CallActor here, instead of waiting for the actual Status persistence to happen. 
If I understand, the reason there is a retry/ack mechanism in the first place is to avoid calling this method before the call has been persisted to "Running", while still ensuring that it will be processed eventually, which makes sense.
But once this message goes through, and the `handleCallCompleted` is called, I don't see why we need to wait for the WA to do its persistence dance (outputs and then status) before Acking to the CA. The WA could potentially do all kind of things / verification etc.. when processing the `CallCompleted` message, but the CA only cares to make sure that this processing happens, it doesn't really need to wait for it to complete.